xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/user/impl/xwiki/MyFormAuthenticator.java

```diff
@@ -292,4 +317,45 @@ public boolean processLogout(SecurityRequestWrapper securityRequestWrapper,
         }
         return result;
     }
+
+    private ObservationManager getObservationManager() {
+        if ( this.observationManager == null ) {
+            this.observationManager = Utils.getComponent(ObservationManager.class);
+        }
+        return this.observationManager;
+    }
+
+    private DocumentReferenceResolver<String> getDocumentReferenceResolver() {
+        if ( this.documentReferenceResolver == null ) {
+            this.documentReferenceResolver =
+                Utils.getComponent(DocumentReferenceResolver.TYPE_STRING, "current");
+        }
+        return this.documentReferenceResolver;
+    }
+
+    private UserReferenceResolver<DocumentReference> getUserReferenceResolver() {
+        if ( this.userReferenceResolver == null ) {
+            this.userReferenceResolver =
+                Utils.getComponent(new DefaultParameterizedType(UserReferenceResolver.class,DocumentReference.class),
+                    "current");
+        }
+        return this.userReferenceResolver;
+    }
+
+    private UserReference getContextUserReference(Principal principal)
+    {
+        UserReference contextUserReference;
+
+        if (principal != null) {
+            // Ensures that the wiki part is removed if specified in the Principal name and if it's not the same wiki
+            // as the current wiki.
+            DocumentReference userDocumentReference =
+                getDocumentReferenceResolver().resolve(principal.getName());
+            contextUserReference = getUserReferenceResolver().resolve(userDocumentReference);
```

Comment:
So here since you want to get the UserReference from  a String directly (principal.getName()), you don't need to go through the DocumentReference resolution first: you can directly use a `UserReferenceResolver<String>`. This is actually even better since `UserReferenceResolver<DocumentReference>` should not really be used.